libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -361,7 +366,7 @@ bool ExpressionCompiler::visit(BinaryOperation const& _binaryOperation)
 	else
 	{
 		bool cleanupNeeded = false;
-		if (Token::isCompareOp(c_op))
+		if (Token::isCompareOp(c_op) || Token::isShiftOp(c_op))
```

Comment:
Strictly, we do not need to clean the left value in case of a left shift, but it is probably better to just clean everything.